From: Vishesh Handa <me@vhanda.in>
Date: Tue, 14 Oct 2014 14:01:40 +0000
Subject: Baloo Runner: Port to newer api
X-Git-Url: http://quickgit.kde.org/?p=plasma-workspace.git&a=commitdiff&h=c0f083dd22e7b5f7d989772a2bbe638c74c1fa16
---
Baloo Runner: Port to newer api

Also, Baloo no longer provides Email results.

The Email parts are now completely separate. We can make the
Akonadi-Search module install a runner for searching through
Emails.
---


--- a/runners/baloo/CMakeLists.txt
+++ b/runners/baloo/CMakeLists.txt
@@ -6,7 +6,7 @@
   KF5::Runner
   KF5::KIOWidgets
   KF5::I18n
-  KF5::BalooCore
+  KF5::Baloo
 )
 
 install(

--- a/runners/baloo/baloosearchrunner.cpp
+++ b/runners/baloo/baloosearchrunner.cpp
@@ -26,6 +26,7 @@
 #include <KRun>
 #include <KRunner/QueryMatch>
 #include <KLocalizedString>
+#include <QMimeDatabase>
 
 #include <Baloo/Query>
 #include <Baloo/Result>
@@ -57,8 +58,7 @@
          << i18n("Image")
          << i18n("Document")
          << i18n("Video")
-         << i18n("Folder")
-         << i18n("Email");
+         << i18n("Folder");
 
     return list;
 }
@@ -75,8 +75,6 @@
         return QIcon::fromTheme("video");
     } else if (category == i18n("Folder")) {
         return QIcon::fromTheme("folder");
-    } else if (category == i18n("Email")) {
-        return QIcon::fromTheme("mail-message");
     }
 
     return Plasma::AbstractRunner::categoryIcon(category);
@@ -102,20 +100,23 @@
     int relevance = 100;
     while (context.isValid() && it.next()) {
         Plasma::QueryMatch match(this);
-        match.setIcon(QIcon::fromTheme(it.icon()));
+        const QUrl url = it.url();
+        QString localUrl = url.toLocalFile();
+
+        QString iconName = QMimeDatabase().mimeTypeForFile(localUrl).iconName();
+        match.setIcon(QIcon::fromTheme(iconName));
         match.setId(it.id());
-        match.setText(it.text());
-        match.setData(it.url());
+        match.setText(url.fileName());
+        match.setData(url);
         match.setType(Plasma::QueryMatch::PossibleMatch);
         match.setMatchCategory(category);
         match.setRelevance(relevance * .01);
         relevance--;
 
-        QString url = it.url().toLocalFile();
-        if (url.startsWith(QDir::homePath())) {
-            url.replace(0, QDir::homePath().length(), QLatin1String("~"));
+        if (localUrl.startsWith(QDir::homePath())) {
+            localUrl.replace(0, QDir::homePath().length(), QLatin1String("~"));
         }
-        match.setSubtext(url);
+        match.setSubtext(localUrl);
 
         context.addMatch(match);
     }
@@ -128,7 +129,6 @@
     match(context, QLatin1String("File/Document"), i18n("Document"));
     match(context, QLatin1String("File/Video"), i18n("Video"));
     match(context, QLatin1String("File/Folder"), i18n("Folder"));
-    match(context, QLatin1String("Email"), i18n("Email"));
 }
 
 void SearchRunner::run(const Plasma::RunnerContext&, const Plasma::QueryMatch& match)

